src/sgp/FileMan.cc

```diff
@@ -1,344 +1,63 @@
-		{
-			case FILE_SEEK_FROM_START: success = distance >= 0 && VfsFile_seekFromStart(f->u.vfile, static_cast<uint64_t>(distance), nullptr); break;
-			case FILE_SEEK_FROM_END:   success = VfsFile_seekFromEnd(f->u.vfile, distance, nullptr); break;
-			default:                   success = VfsFile_seekFromCurrent(f->u.vfile, distance, nullptr); break;
-		}
-	}
-	if (!success) throw std::runtime_error("Seek in file failed");
-}
-
-
-INT32 FileGetPos(const SGPFile* f)
-{
-	bool success;
-	uint64_t position = 0;
-	if (f->flags & SGPFILE_REAL)
-	{
-		position = File_seekFromCurrent(f->u.file, 0);
-		success = position != UINT64_MAX;
-	}
-	else
-	{
-		success = VfsFile_seekFromCurrent(f->u.vfile, 0, &position);
-	}
-
-	if (!success)
-	{
-		RustPointer<char> err{getRustError()};
-		STLOGE("FileGetPos: {}", err.get());
-		throw std::runtime_error("Getting file position failed");
-	}
-	if (position > INT32_MAX)
-	{
-		STLOGW("FileGetPos: truncating from {} to {}", position, INT32_MAX);
-		position = INT32_MAX;
-	}
-	return static_cast<INT32>(position);
-}
-
-
-UINT32 FileGetSize(const SGPFile* f)
-{
-	bool success;
-	uint64_t len = 0;
-	if (f->flags & SGPFILE_REAL)
-	{
-		len = File_len(f->u.file);
-		success = len != UINT64_MAX;
-	}
-	else
-	{
-		success = VfsFile_len(f->u.vfile, &len);
-	}
-
-	if (!success)
-	{
-		RustPointer<char> err{getRustError()};
-		STLOGE("FileGetSize: {}", err.get());
-		throw std::runtime_error("Getting file size failed");
-	}
-	if (len > UINT32_MAX)
-	{
-		STLOGW("FileGetSize: truncating from {} to {}", len, UINT32_MAX);
-		len = UINT32_MAX;
-	}
-	return static_cast<UINT32>(len);
-}
-
-
 void FileMan::createDir(const ST::string& path)
 {
 	if (!Fs_isDir(path.c_str()) && !Fs_createDir(path.c_str()))
 	{
 		RustPointer<char> err{getRustError()};
-		STLOGE("Failed to created directory '{}': {}", path, err.get());
-		throw std::runtime_error("Failed to create directory");
+		throw IoException(ST::format("FileMan::createDir('{}') failed: {}", path, err.get()));
 	}
 }
 
 
-void EraseDirectory(const ST::string& dirPath)
+void FileMan::eraseDir(const ST::string& dirPath)
 {
-	std::vector<ST::string> paths = FindAllFilesInDir(dirPath);
+	std::vector<ST::string> paths = FileMan::findAllFilesInDir(dirPath);
 	for (const ST::string& path : paths)
 	{
+		if (FileMan::isDir(path)) continue;
 		try
 		{
-			FileDelete(path);
+			FileMan::deleteFile(path);
 		}
-		catch (const std::runtime_error& ex)
+		catch (const IoException& ex)
 		{
-			if (Fs_isDir(path.c_str())) continue;
-			STLOGE("EraseDirectory '{}' '{}': {}", dirPath, path, ex.what());
-			throw;
+			throw IoException(ST::format("FileMan::eraseDir('{}') failed because {}", dirPath, ex.what()));;
 		}
 	}
 }
 
-
-File* GetRealFileHandleFromFileManFileHandle(const SGPFile* f)
-{
-	return f->flags & SGPFILE_REAL ? f->u.file : nullptr;
-}
-
-uint64_t GetFreeSpaceOnHardDriveWhereGameIsRunningFrom(void)
+uint64_t FileMan::getFreeSpaceOnHardDriveWhereGameIsRunningFrom(void)
```

Comment:
Out of scope of this PR and probably only on rare cases - 

This check is not accurate. We should check the disk free on temp and userPrivate directories, not the game executable.

Reply:
Is this the purpose / side-effect of the `FileMan::switchTmpFolder` line that was just removed?

Reply:
Good catch. Yes before it checked the user directory, with `switchTmpFolder`. I added a commit that adapts this check to check for min(userDir, tempDir).